doc/user_guide/psyir.rst

```diff
@@ -100,6 +100,56 @@ subclassed into a kernel-layer ``KernelSchedule``. In addition to
 ``BinaryOperation`` and ``NaryOperation``.
 
 
+Node Descriptions
+=================
+
+The Range node
```

Comment:
As discussed at the telco, please could you move this description into the docstring of the `Range` class. I suspect we then don't need to mention it here at all. Perhaps add a sentence pointing people to the reference guide for the definitions of the various nodes (https://psyclone-ref.readthedocs.io/en/latest/autogenerated/psyclone.psyir.nodes.html#classes)

Reply:
Moved text into docstring. I didn't add a link to the reference guide as the user guide documentation is essentially the same in this case.

Reply:
My intention was that we'd leave all detailed node descriptions to the Reference Guide from now on. I thought we'd agreed that but, as usual, I guess we actually agreed on something slightly different :-)
Anyway, I don't think it warrants holding up this PR any more so I'll leave as is for the moment.

Reply:
Yes I think we did agree that :-).